lib/ostruct.rb

```diff
@@ -220,27 +235,33 @@ def freeze
     len = args.length
     if mname = mid[/.*(?==\z)/m]
       if len != 1
-        raise ArgumentError, "wrong number of arguments (given #{len}, expected 1)", caller(1)
+        raise! ArgumentError, "wrong number of arguments (given #{len}, expected 1)", caller(1)
       end
       set_ostruct_member_value!(mname, args[0])
-    elsif len == 0
-    elsif @table.key?(mid)
-      raise ArgumentError, "wrong number of arguments (given #{len}, expected 0)"
     else
+      if len == 0
+        respond_to_unknown_attribute!(mname) { |val| return val }
```

Comment:
Won't `mname` be nil here?

```suggestion
        respond_to_unknown_attribute!(mid) { |val| return val }
```

Reply:
Indeed. Clearly I'm missing a test 😅